File: Python/src/algorithm/FlatMap.py
def flat_map(array):
    res = []
    for item in array:
        if type(item) == list:
            res = [*res, *flat_map(item)]  # unpacking list with *
        else:
            res.append(item)
    return res


def main():
    arr = [1, [2, 3], [4, [5, [6, 7], 8]], 9, [10]]
    print(flat_map(arr))


if __name__ == "__main__":
    main()

File: src/main/java/algorithm/FlatMap.java
package algorithm;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.logging.Logger;

public class FlatMap {

    public static void main(String[] args) {
        Logger logger = Logger.getLogger(FlatMap.class.getName());
        Object[] arr = new Object[]{1, 2, new Object[]{3, new Object[]{4, 5}}, new Object[]{6, new Object[]{7, 8, new Object[]{9, 10}}}};

        String res = flattenMap(arr).toString();
        logger.info(res);
    }

    private static ArrayList<Integer> flattenMap(Object[] arr) {
        ArrayList<Integer> arrayList = new ArrayList<>();
        Arrays.stream(arr).forEach(item -> {
            if (item instanceof Object[]) {
                arrayList.addAll(flattenMap((Object[]) item));
            } else arrayList.add((int) item);
        });
        return arrayList;
    }
}
